### PR TITLE
Update explained variance metric

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Update to explained variance metric
* Add checks for shape
* Add checks for division by zero
* Adds `multioutput` parameter for handling multioutput cases (same as sklearn)

Also add a bit of description to the new metric testing function

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/metrics/regression/explained_variance.py [hunk 1]
```diff
@@ -43,6 +83,8 @@ def update(self, preds: torch.Tensor, target: torch.Tensor):
             preds: Predictions from model
             target: Ground truth values
         """
+        assert preds.shape == target.shape, \
```
Comment (Borda) [inline pytorch_lightning/metrics/regression/explained_variance.py:86]: lets consider using Error clases instead...
https://medium.com/@jadhavmanoj/python-what-is-raise-and-assert-statement-c3908697bc62#:~:text=Python's%20assert%20statement%20is%20a,a%20bug%20in%20your%20program.

File: tests/metrics/utils.py [hunk 2]
```diff
@@ -53,6 +57,19 @@ def _compute_batch(rank, preds, target, metric_class, sk_metric, ddp_sync_on_ste
 
 
 def compute_batch(preds, target, metric_class, sk_metric, ddp_sync_on_step, ddp=False, metric_args={}):
```
Comment (Borda) [inline tests/metrics/utils.py:59]: can we also add typing to all these functions?

File: pytorch_lightning/metrics/regression/explained_variance.py (hunk 1 above)
Comment (SkafteNicki) [reply pytorch_lightning/metrics/regression/explained_variance.py:86]: So instead `assert` it should be a `raise`?
